### ISSUE TITLE
Run Functional Tests task fails with System.ArgumentException

### ISSUE THREAD

**Issue by nesshoever** [Microsoft/vsts-tasks#5677]
I'm not sure whether the error is caused by a recent update on the VSTS side or by changes on our side, but at least the error message points to code in a utility class in the TestManagement.Server namespace.

## Environment
Running on VSTS
Account name: arivis-neuron
Team project name: neuron-platform
Build definition: [CI] Build, publish, deploy & run UI test
Build number: 20171020.2
Build Id: 3102
Agent: Hosted VS2017

## Issue Description
The test fails with a System.ArgumentException for the AutomatedTestName in task 'Run tests':
```
2017-10-20T07:13:30.7267096Z ##[warning]DistributedTests: Test run is aborted. Logging details of the run logs.
2017-10-20T07:13:30.7267096Z ##[debug]Processed: ##vso[task.logissue type=warning]DistributedTests: Test run is aborted. Logging details of the run logs.
2017-10-20T07:13:30.8103352Z ##[warning]DistributedTests: New test run created.
2017-10-20T07:13:30.8103352Z ##[debug]Processed: ##vso[task.logissue type=warning]DistributedTests: New test run created.
2017-10-20T07:13:30.8103352Z Test Run queued for Project Collection Build Service (arivis-neuron).
2017-10-20T07:13:30.8103352Z 
2017-10-20T07:13:30.8103352Z ##[warning]DistributedTests: Error in slicing for TestRun.0.\nSystem.ArgumentException: The string must have at least one character.
2017-10-20T07:13:30.8113364Z ##[debug]Processed: ##vso[task.logissue type=warning]DistributedTests: Error in slicing for TestRun.0.\nSystem.ArgumentException: The string must have at least one character.
2017-10-20T07:13:30.8113364Z Parameter name: AutomatedTestName
2017-10-20T07:13:30.8113364Z    at Microsoft.TeamFoundation.TestManagement.Server.SliceUtility.CreateSliceFromTestCaseResult(List`1 testCaseResults) in D:\v2.0\A1\_work\14\s\Tfs\Service\TestManagement\TestExecutionService\JobPlugins\Slicing\SliceUtility.cs:line 228
2017-10-20T07:13:30.8113364Z    at Microsoft.TeamFoundation.TestManagement.Server.SliceUtility.CreateDiscoveryDataForRun(IVssRequestContext requestContext, WorkflowJobData workFlowJobData) in D:\v2.0\A1\_work\14\s\Tfs\Service\TestManagement\TestExecutionService\JobPlugins\Slicing\SliceUtility.cs:line 219
2017-10-20T07:13:30.8113364Z    at Microsoft.TeamFoundation.TestManagement.Server.SliceUtility.TryPerformSlicingFromTestPoints(IVssRequestContext requestContext, WorkflowJobData workFlowJobData, String& resultMessage, ISlicingEngine slicingEngine) in D:\v2.0\A1\_work\14\s\Tfs\Service\TestManagement\TestExecutionService\JobPlugins\Slicing\SliceUtility.cs:line 98
2017-10-20T07:13:30.8113364Z ##[warning]DistributedTests: Test run aborted. Test run id: 1325
2017-10-20T07:13:30.8113364Z ##[debug]Processed: ##vso[task.logissue type=warning]DistributedTests: Test run aborted. Test run id: 1325
2017-10-20T07:13:30.8123367Z ##[error]System.Exception: The test run was aborted, failing the task.
2017-10-20T07:13:30.8123367Z ##[debug]Processed: ##vso[task.logissue type=error;]System.Exception: The test run was aborted, failing the task.
2017-10-20T07:13:30.8709617Z ##[debug]DistributedTests: Deleted test settings with id : 296
2017-10-20T07:13:30.9152488Z ##[error]PowerShell script completed with 1 errors.
2017-10-20T07:13:30.9159590Z ##[section]Finishing: Run Tests
```

## Error logs
[logs_3102.zip](https://github.com/Microsoft/vsts-tasks/files/1401039/logs_3102.zip)

**Comment by nesshoever** [comment]
After digging further into this issue I finaly found that the error was caused by a test case that had an abmiguous 'automation status'. Instead of either 'Not automated' or 'Planned' it was just empty.
This caused the test case to be included into a test run, because all other parameters (like configuration, etc.) made it eligible. I guess because there was not data to link it to any automation assembly/method, this is where the "System.ArgumentException: The string must have at least one character." is caused.
So basically this is a configuration error, but it would be nice, if the task would be more robust and simply skip this test with a message rather than fail with an exception.
